### PR TITLE
Support Time-to-live in query(syntax)

### PR DESCRIPTION
**Pull request by steppenwolfyuetong**
As title.

### CODE REVIEW

**Comment by steppenwolfyuetong** [comment]
Rebase and please review，thx.

File: src/parser/MaintainSentences.cpp [hunk 1]
```diff
@@ -9,6 +9,55 @@
...
+std::string SchemaPropItem::toString() const {
+    switch (propType_) {
+        case TTL_DURATION:
+            return folly::stringPrintf("ttl_duration = %ld",
+                                       boost::get<int64_t>(propValue_));
+        case TTL_COL:
+            return folly::stringPrintf("ttl_col = %s",
+                                       boost::get<std::string>(propValue_).c_str());
+        // TODO(YT) The following features will be supported in the future
+        case COMMENT:
```
Comment (dangleptr) [inline src/parser/MaintainSentences.cpp]: We'd better remove them before discussing what configurations we should support.

File: src/parser/MaintainSentences.h [hunk 2]
```diff
@@ -76,11 +59,183 @@ class ColumnSpecificationList final {
 };
 
 
+class SchemaPropItem final {
+public:
+    using Value = boost::variant<int64_t, bool, std::string>;
+
+    enum PropType : uint8_t {
```
Comment (dangleptr) [inline src/parser/MaintainSentences.h]: ditto

File: src/parser/scanner.lex [hunk 3]
```diff
@@ -95,6 +94,14 @@ ON                          ([Oo][Nn])
 ROLES                       ([Rr][Oo][Ll][Ee][Ss])
 BY                          ([Bb][Yy])
 IN                          ([Ii][Nn])
+TTL_DURATION                ([Tt][Tt][Ll][_][Dd][Uu][Rr][Aa][Tt][Ii][Oo][Nn])
+TTL_COL                     ([Tt][Tt][Ll][_][Cc][Oo][Ll])
+COMMENT                     ([Cc][Oo][Mm][Mm][Ee][Nn][Tt])
```
Comment (dangleptr) [inline src/parser/scanner.lex]: ditto

File: src/parser/test/ScannerTest.cpp [hunk 4]
```diff
@@ -313,6 +311,30 @@ TEST(Scanner, Basic) {
         CHECK_SEMANTIC_TYPE("by", TokenType::KW_BY),
         CHECK_SEMANTIC_TYPE("IN", TokenType::KW_IN),
         CHECK_SEMANTIC_TYPE("In", TokenType::KW_IN),
+        CHECK_SEMANTIC_TYPE("TTL_DURATION", TokenType::KW_TTL_DURATION),
+        CHECK_SEMANTIC_TYPE("ttl_duration", TokenType::KW_TTL_DURATION),
+        CHECK_SEMANTIC_TYPE("Ttl_duration", TokenType::KW_TTL_DURATION),
+        CHECK_SEMANTIC_TYPE("TTL_COL", TokenType::KW_TTL_COL),
+        CHECK_SEMANTIC_TYPE("ttl_col", TokenType::KW_TTL_COL),
+        CHECK_SEMANTIC_TYPE("Ttl_col", TokenType::KW_TTL_COL),
+        CHECK_SEMANTIC_TYPE("COMMENT", TokenType::KW_COMMENT),
```
Comment (dangleptr) [inline src/parser/test/ScannerTest.cpp]: ditto

File: src/parser/MaintainSentences.cpp (hunk 1 above)
Comment (steppenwolfyuetong) [reply src/parser/MaintainSentences.cpp]: Yeah，I will remove them.
Written here only identifies them as attributes of the schema

File: src/parser/MaintainSentences.h (hunk 2 above)
Comment (steppenwolfyuetong) [reply src/parser/MaintainSentences.h]: 👍

File: src/parser/scanner.lex (hunk 3 above)
Comment (steppenwolfyuetong) [reply src/parser/scanner.lex]: 👍

File: src/parser/test/ScannerTest.cpp (hunk 4 above)
Comment (steppenwolfyuetong) [reply src/parser/test/ScannerTest.cpp]: 👍

**Comment by steppenwolfyuetong** [comment]
Fix comments and rebase master, please review again, thx.

File: src/parser/MaintainSentences.h [hunk 5]
```diff
@@ -76,11 +59,123 @@ class ColumnSpecificationList final {
...
+        propType_ = op;
+        propValue_ = val;
+    }
+
+    SchemaPropItem(PropType op, std::string val) {
+        propType_ = op;
+        propValue_ = std::move(val);
+    }
+
+    int64_t asInt() {
```
Comment (dangleptr) [inline src/parser/MaintainSentences.h]: If you don't want to expose the methods to outside, just make them private.

File: src/parser/MaintainSentences.h [hunk 6]
```diff
@@ -76,11 +59,123 @@ class ColumnSpecificationList final {
...
+        return boost::get<int64_t>(propValue_);
+    }
+
+    std::string& asString() {
```
Comment (dangleptr) [inline src/parser/MaintainSentences.h]: Return reference?

File: src/parser/MaintainSentences.h [hunk 7]
```diff
@@ -76,11 +59,123 @@ class ColumnSpecificationList final {
...
+
+    bool isBool() {
+        return propValue_.which() == 1;
+    }
+
+    bool isString() {
+        return propValue_.which() == 2;
+    }
+
+    int64_t getTtlDuration() {
```
Comment (dangleptr) [inline src/parser/MaintainSentences.h]: StatusOr?

File: src/parser/MaintainSentences.h [hunk 8]
```diff
@@ -76,11 +59,123 @@ class ColumnSpecificationList final {
...
+        if (isInt()) {
+            return asInt();
+        } else {
+            LOG(ERROR) << "Ttl_duration value illegal.";
+            return 0;
+        }
+    }
+
+    std::string getTtlCol() {
```
Comment (dangleptr) [inline src/parser/MaintainSentences.h]: ditto

File: src/parser/MaintainSentences.h (hunk 5 above)
Comment (steppenwolfyuetong) [reply src/parser/MaintainSentences.h]: 👍，good catch!

File: src/parser/MaintainSentences.h (hunk 8 above)
Comment (steppenwolfyuetong) [reply src/parser/MaintainSentences.h]: Good point, I will do it.

File: src/parser/MaintainSentences.h (hunk 7 above)
Comment (steppenwolfyuetong) [reply src/parser/MaintainSentences.h]: Good point, I will do it.

**Comment by steppenwolfyuetong** [comment]
Fix comments and rebase master, please review again, thx.

**Comment by dangleptr** [comment]
Jenkins go

**Comment by steppenwolfyuetong** [comment]
Rebase master

**Comment by dangleptr** [comment]
Jenkins go

File: src/parser/MaintainSentences.h [hunk 9]
```diff
@@ -76,11 +59,124 @@ class ColumnSpecificationList final {
...
+private:
+    int64_t asInt() {
+        return boost::get<int64_t>(propValue_);
+    }
+
+    const std::string& asString() {
+        return boost::get<std::string>(propValue_);
+    }
+
+    bool asBool() {
```
Comment (laura-ding) [inline src/parser/MaintainSentences.h]: why need this？

File: src/parser/MaintainSentences.h [hunk 10]
```diff
@@ -76,11 +59,124 @@ class ColumnSpecificationList final {
...
+                DCHECK(false);
+        }
+        return false;
+    }
+
+    bool isInt() {
+        return propValue_.which() == 0;
+    }
+
+    bool isBool() {
```
Comment (laura-ding) [inline src/parser/MaintainSentences.h]: ditto

File: src/parser/MaintainSentences.h (hunk 9 above)
Comment (steppenwolfyuetong) [reply src/parser/MaintainSentences.h]: Because the type of some values can be bool type.

File: src/parser/MaintainSentences.h (hunk 10 above)
Comment (steppenwolfyuetong) [reply src/parser/MaintainSentences.h]: Because the type of some values can be bool type.

**Pull request merged**